src/Microsoft.DotNet.ProjectModel.Server/InternalModels/ProjectContextSnapshot.cs

```diff
@@ -24,31 +25,38 @@ internal class ProjectContextSnapshot
         public static ProjectContextSnapshot Create(ProjectContext context, string configuration, IEnumerable<string> currentSearchPaths)
         {
             var snapshot = new ProjectContextSnapshot();
-            
+
             var allDependencyDiagnostics = new List<DiagnosticMessage>();
             allDependencyDiagnostics.AddRange(context.LibraryManager.GetAllDiagnostics());
             allDependencyDiagnostics.AddRange(DependencyTypeChangeFinder.Diagnose(context, currentSearchPaths));
 
             var diagnosticsLookup = allDependencyDiagnostics.ToLookup(d => d.Source);
 
+            var allExports = context.CreateExporter(configuration).GetAllExports();
             var allSourceFiles = new List<string>(context.ProjectFile.Files.SourceFiles);
             var allFileReferences = new List<string>();
             var allProjectReferences = new List<ProjectReferenceDescription>();
             var allDependencies = new Dictionary<string, DependencyDescription>();
-            
-            foreach (var export in context.CreateExporter(configuration).GetDependencies())
+
+            var frameworkExports = allExports.Where(export => export.Library.Identity.Type == LibraryType.ReferenceAssembly)
+                                             .ToDictionary(export => export.Library.Identity.Name);
```

Comment:
Looks like you always end up having to strip off the `fx/` when looking things up in this dictionary. How about you apply the prefix to the keys here so you don't have to keep doing that?